public/app/features/trails/otel/utils.test.ts

```diff
@@ -270,3 +277,42 @@ describe('updateOtelJoinWithGroupLeft', () => {
     expect(otelJoinQueryVar.getValue()).toBe('');
   });
 });
+
+describe('getProdOrDefaultOption', () => {
+  it('should return the value of the option containing "prod"', () => {
+    const options = [
+      { value: 'test1', label: 'Test 1' },
+      { value: 'prod2', label: 'Prod 2' },
+      { value: 'test3', label: 'Test 3' },
+    ];
+    expect(getProdOrDefaultOption(options)).toBe('prod2');
+  });
+
+  it('should return the first option value if no option contains "prod"', () => {
+    const options = [
+      { value: 'test1', label: 'Test 1' },
+      { value: 'test2', label: 'Test 2' },
+      { value: 'test3', label: 'Test 3' },
+    ];
+    expect(getProdOrDefaultOption(options)).toBe('test1');
+  });
+
+  it('should handle case insensitivity', () => {
+    const options = [
+      { value: 'test1', label: 'Test 1' },
+      { value: 'PROD2', label: 'Prod 2' },
+      { value: 'test3', label: 'Test 3' },
+    ];
+    expect(getProdOrDefaultOption(options)).toBe('PROD2');
+  });
+
+  it('should return the first option value if the options array is empty', () => {
```

Comment:
The `it('')` doesn't yet match what's tested here:

```suggestion
  it('should throw if the options array is empty', () => {
```

Reply:
Returning null when options are empty instead of throwing the error, fixed the test to reflect this